### PR TITLE
make Snackbar, PeoplePickerView, CalendarView, DatePicker Duo compliant

### PR DESCRIPTION
**Pull request by aishwarya56**
### Platforms Impacted
- [Android

### Description of changes

We have added support for duo devices in the fluent library. By these changes Snackbar, DateTimepicker, CalendarView and peoplePickerView will be Duo compliant. 
Anything under the hindge will be shifted to left or right depending on scenarios

| Before                                       | After                                      |
|----------------------------------------------|--------------------------------------------|
|device-2020-11-05-130928|device-2020-11-05-130404|
|device-2020-11-05-130955|device-2020-11-05-130512|
|device-2020-11-05-131156|device-2020-11-05-130540|

### CODE REVIEW

File: FluentUI/src/main/java/com/microsoft/fluentui/util/DuoSupportUtils.kt [hunk 1]
```diff
@@ -12,38 +13,38 @@ import java.lang.Exception
  */
 object DuoSupportUtils {
     const val DUO_HINGE_WIDTH = 84
+    const val COLUMNS_IN_START_DUO_MODE = 3
+    const val COLUMNS_IN_END_DUO_MODE = 4
 
     @JvmStatic
-    fun isDeviceSurfaceDuo(activity: Activity) = ScreenHelper.isDeviceSurfaceDuo(activity)
+    fun isDeviceSurfaceDuo(activity: AppCompatActivity) = ScreenHelper.isDeviceSurfaceDuo(activity)
```
Comment (mnkshukl) [inline FluentUI/src/main/java/com/microsoft/fluentui/util/DuoSupportUtils.kt:20]: is AppCompatActivity required or Activity is enough as some old base apps  might not be using Appcompat yet.
I know office has just shifted

File: FluentUI/src/main/java/com/microsoft/fluentui/calendar/WeekHeadingView.kt [hunk 2]
```diff
@@ -47,7 +51,27 @@ internal class WeekHeadingView : LinearLayout {
...
+                                addView(View(context), LayoutParams(0, LayoutParams.MATCH_PARENT, (DuoSupportUtils.getHingeWidth(it).toFloat())))
+                            }
+                            else -> {
+                                addView(textView, LayoutParams(0, LayoutParams.MATCH_PARENT, (context.displaySize.x - DuoSupportUtils.getHingeWidth(it)) / 8.0f))
+
+                            }
+                        }
+                    } else {
+                        addView(textView, LayoutParams(0, LayoutParams.MATCH_PARENT, 1.0f))
+                    }
```
[the remark this answers is not in the document]
Comment (aishwarya56) [reply FluentUI/src/main/java/com/microsoft/fluentui/calendar/WeekHeadingView.kt:72]: fixed

File: FluentUI/src/main/java/com/microsoft/fluentui/calendar/WeeksView.kt [hunk 3]
```diff
@@ -115,6 +119,15 @@ internal class WeeksView : MSRecyclerView {
         setHasFixedSize(true)
         layoutManager = GridLayoutManager(context, DAYS_IN_WEEK, LinearLayoutManager.VERTICAL, false)
         layoutManager?.scrollToPosition(pickerAdapter.todayPosition)
+        post {
```
[the remark this answers is not in the document]
Comment (aishwarya56) [reply FluentUI/src/main/java/com/microsoft/fluentui/calendar/WeeksView.kt:122]: Can be removed, thanks for pointing it out

[the remark this answers is not in the document]
Comment (aishwarya56) [reply FluentUI/src/main/java/com/microsoft/fluentui/calendar/WeeksView.kt:122]: Intersecthinge works when the view is visible,
Post method move it to the ui thread queue and hence it works as per expectation

File: FluentUI/src/main/java/com/microsoft/fluentui/util/DuoSupportUtils.kt (hunk 1 above)
Comment (aishwarya56) [reply FluentUI/src/main/java/com/microsoft/fluentui/util/DuoSupportUtils.kt:20]: Using activity now

File: FluentUI/src/main/java/com/microsoft/fluentui/peoplepicker/PeoplePickerTextView.kt [hunk 4]
```diff
@@ -411,6 +433,35 @@ internal class PeoplePickerTextView : TokenCompleteTextView<IPersona> {
...
         text.insert(offset, spannableStringBuilder)
         text.setSpan(personaSpan, offset, offset + spannableStringBuilder.length - 1, Spanned.SPAN_EXCLUSIVE_EXCLUSIVE)
+        context.activity?.let {
+            if (DuoSupportUtils.isDualScreenMode(it)) {
+                checkForIntersectionWithHinge(personaSpan)
+            }
+        }
+    }
+
+    private fun checkForIntersectionWithHinge(tokenImageSpan: TokenImageSpan) {
```
[the remark this answers is not in the document]
Comment (aishwarya56) [reply FluentUI/src/main/java/com/microsoft/fluentui/peoplepicker/PeoplePickerTextView.kt:443]: Created a bug for the same, will handle this as a bug and implement later on
https://dev.azure.com/microsoftdesign/fluentui-native/_workitems/edit/10743

File: FluentUI/src/main/java/com/microsoft/fluentui/calendar/CalendarAdapter.kt [hunk 5]
```diff
@@ -171,6 +172,10 @@ internal class CalendarAdapter : RecyclerView.Adapter<CalendarAdapter.CalendarDa
         }
     }
 
+    fun setViewHeight(rowHeight: Int) {
+        viewHeight = rowHeight
+    }
```
[the remark this answers is not in the document]
Comment (aishwarya56) [reply FluentUI/src/main/java/com/microsoft/fluentui/calendar/CalendarAdapter.kt]: Moved to constructor

**Review by mnkshukl — APPROVED** [review]

**Pull request merged**